client/src/main/java/se/fortnox/reactivewizard/client/ConcatenatedJsonObjectsDeserializer.java

```diff
@@ -0,0 +1,123 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.Sinks;
+import se.fortnox.reactivewizard.util.ReflectionUtil;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Deque;
+import java.util.LinkedList;
+import java.util.List;
+
+import static java.lang.String.format;
+
+public class ConcatenatedJsonObjectsDeserializer {
+    private final Deque<Byte> tokenStack;
+    private final ObjectReader reader;
+    private String currentJsonObject;
+
+
+    public ConcatenatedJsonObjectsDeserializer(ObjectMapper objectMapper, Method method) {
+        tokenStack = new ArrayDeque<>();
+        currentJsonObject = "";
+        Type type = ReflectionUtil.getTypeOfObservable(method);
+        JavaType javaType = TypeFactory.defaultInstance().constructType(type);
+        reader   = objectMapper.readerFor(javaType);
+    }
+
+    public Publisher<?> process(byte[] responseBodyChunk) {
+        if (responseBodyChunk.length == 0) {
+            return Mono.empty();
+        }
+
+        int startParsingFrom = 0;
+        byte[] contentToAppend;
+        List<Object> nextObjects = new LinkedList<>();
+        for (int i = 0; i < responseBodyChunk.length; i++) {
+            byte currentByte = responseBodyChunk[i];
+
+            if (tokenStack.isEmpty() && !isStartingBracket(currentByte) && !isCitation(currentByte)) {
+                return Mono.error(new ConcatenatedJsonObjectDeserializerException("Failed to deserialize concatenated json objects, expected '{' or '\"' found '%c'", currentByte));
+            } else if (tokenStack.isEmpty() && (isStartingBracket(currentByte) || isCitation(currentByte))) {
+                tokenStack.push(currentByte);
+                continue;
+            }
+
+            assert !tokenStack.isEmpty();
+
+            if (isStartingBracket(currentByte) && !isCitation(tokenStack.peek())) {
+                tokenStack.push(currentByte);
+            } else if (isClosingBracket(currentByte) && !isCitation(tokenStack.peek())) {
+                tokenStack.pop();
+            } else if (isCitation(currentByte) && isCitation(tokenStack.peek()) && !isEscaped()) {
+                tokenStack.pop();
+            } else if (isCitation(currentByte) && !isCitation(tokenStack.peek()) && !isEscaped()) {
+                tokenStack.push(currentByte);
+            }
+
+            if (tokenStack.isEmpty()) {
+                contentToAppend = Arrays.copyOfRange(responseBodyChunk, startParsingFrom, i + 1);
+                currentJsonObject = currentJsonObject.concat(new String(contentToAppend, StandardCharsets.UTF_8)).trim();
```

Comment:
We could just push the bytes onto a buffer with a type that fits readValue - that would avoid allocating strings.

Reply:
Good point. Perhaps we could have gotten a better solution here if JsonParser was used. That said, I deleted ConcatenatedJsonObjectDeserializer alltogether.